crates/worker/src/partition_processor_manager/spawn_processor_task.rs

```diff
@@ -233,3 +176,185 @@ impl SpawnPartitionProcessorTask {
+) -> anyhow::Result<PartitionStore> {
+    let partition_store_exists = partition_store_manager
+        .has_partition_store(partition_id)
+        .await;
+
+    if partition_store_exists && fast_forward_lsn.is_none() {
+        // We have an initialized partition store, and no fast-forward target - go on and open it.
+        Ok(partition_store_manager
+            .open_partition_store(
+                partition_id,
+                key_range,
+                OpenMode::OpenExisting,
+                &options.storage.rocksdb,
+            )
+            .await?)
+    } else {
+        // We either don't have an existing local partition store initialized - or we have a
+        // fast-forward LSN target for the local state (probably due to seeing a log trim-gap).
+        Ok(create_or_recreate_store(
+            partition_id,
+            partition_store_manager,
+            snapshot_repository,
+            fast_forward_lsn,
+            &options,
+            key_range,
+        )
+        .await?)
+    }
+}
+
+/// (Re-)creates a fresh partition store based on the available snapshot and an optional
+/// fast-forward LSN. Assumes that the partition store does not yet exist, unless a fast-forward
+/// LSN is set. If a fast-forward LSN is set, but there is no snapshot repository, this will always
+/// fail. An existing store will be dropped only if a snapshot is found in the repository with an
+/// LSN greater than the fast-forward target.
+async fn create_or_recreate_store(
+    partition_id: PartitionId,
+    partition_store_manager: PartitionStoreManager,
+    snapshot_repository: Option<SnapshotRepository>,
+    fast_forward_lsn: Option<Lsn>,
+    options: &&WorkerOptions,
+    key_range: RangeInclusive<PartitionKey>,
+) -> anyhow::Result<PartitionStore> {
+    // Attempt to get the latest available snapshot from the snapshot repository:
+    let snapshot = match snapshot_repository {
+        Some(repository) => {
+            debug!("Looking for partition snapshot from which to bootstrap partition store");
+            // todo(pavel): pass target LSN to repository
+            repository.get_latest(partition_id).await?
+        }
+        None => {
+            debug!("No snapshot repository configured");
+            None
+        }
+    };
+
+    Ok(match (snapshot, fast_forward_lsn) {
+        (None, None) => {
+            info!("No snapshot found to bootstrap partition, creating new store");
+            partition_store_manager
+                .open_partition_store(
+                    partition_id,
+                    key_range,
+                    OpenMode::CreateIfMissing,
+                    &options.storage.rocksdb,
+                )
+                .await?
+        }
+        (Some(snapshot), None) => {
+            // We only reach this point if there is no initialized store for the partition (early
+            // return at start of method), we can import without first dropping the column family.
```

Comment:
The comment in the parenthesis seems to be no longer correct.

Reply:
Ah! Will update before I merge 👍